### PR TITLE
Remove qt4 dependency from developer packages

### PR DESCRIPTION
**Pull request by peterfpeterson**
Since mantidplot is no longer in the codebase, the dependencies are no longer needed by developers. The deb package is also compatible with ubuntu 20.04 so much of those getting started instructions have been removed.

**To test:**

Build the deb and rpm packages and install them.

*There is no associated issue.*

*This does not require release notes* because it is a change in the developer packages.

The version into `ornl-next` is #32015.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec [hunk 1]
```diff
@@ -71,8 +70,6 @@ Requires: qt5-qtx11extras
 Requires: qt5-qtx11extras-devel
 Requires: qt-devel >= 4.6
```
Comment (martyngigg) [inline buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec:71]: Should we bump this to `> 5`?

Comment (peterfpeterson) [reply buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec:71]: `4.8.7-9.el7_9` is the version we have on our RHEL7 machines. We could probably move to `>= 4.8`.

Comment (martyngigg) [reply buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec:71]: Do we need it? Is it not the qt4 dependencies?

Comment (peterfpeterson) [reply buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec:71]: I've never tried without it

**Review by AndreiSavici — DISMISSED** [review]
Ubuntu 20.04 looks OK

File: buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec (hunk 1 above)
Comment (martyngigg) [reply buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec:71]: So it looks like we can remove `qt-devel`, `qtwebkit-devel` & `qscintilla-devel` and this should completely remove Qt4. I tested a build locally in a docker container and it worked.

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Comment by peterfpeterson** [comment]
For ubuntu, we have `libqscintilla2-qt5-dev` *and* `libqt5scintilla2-dev`. Which of those should be dropped?

**Comment by AndreiSavici** [comment]
Ubuntu 20.04 has `libqscintilla2-qt5-dev`

**Comment by martyngigg** [comment]
I think `libqt5scintilla2-dev` is a virtual package that points to the other one anyway.

**Review by martyngigg — APPROVED** [review]
This looks good to me.

**Pull request merged**